File: Backend/todoapp/todoapp/src/main/java/com/todo/todoapp/service/TodoService.java
package com.todo.todoapp.service;

import com.todo.todoapp.model.Todo;

import java.util.List;

public interface TodoService {

    Todo createTodo(Todo todo);

    List<Todo> getAllTodos(Boolean completed);

    Todo updateTodo(Long id, Todo todo);

    void deleteTodo(Long id);
}

File: Frontend/todo-frontend/src/services/todoService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/todos";

// Get Todos (with optional filter)
export const getTodos = (completed) => {
  if (completed === undefined) {
    return axios.get(API_URL);
  }
  return axios.get(`${API_URL}?completed=${completed}`);
};

export const createTodo = (todo) =>
  axios.post(API_URL, todo);

export const updateTodo = (id, todo) =>
  axios.put(`${API_URL}/${id}`, todo);

export const deleteTodo = (id) =>
  axios.delete(`${API_URL}/${id}`);
